src/entities/Tree.js

```diff
@@ -0,0 +1,76 @@
+import cloneDeep from 'lodash/cloneDeep';
+
+export class Tree {
+  treeData;
+
+  constructor(treeData) {
+    this.treeData = treeData;
+    this.traversalByNode = this.traversalByNode.bind(this);
+    this.updateNode = this.updateNode.bind(this);
+    this.renderTree = this.renderTree.bind(this);
+  }
+
+  traversalByNode(curNode, fn) {
+    const c = this;
```

Comment:
We should use this name, you mean what? "C" in Vietnamese we can understand in many means

Reply:
I think should be `ctx` stands for `context`